Succubus Trait

Raikyr0 opened the pull request:
# Description

Adds the Succubus Trait. It uses reverse-engineered Liquor Lifeline code.
In practice, similar to the Vampire metabolism of blood, but heals less, without the food restriction (in case you don't want to be caught chugging cum in public when someone invites you to eat).

Alright, i've redone alot of shit, instead of using old vampire ideas like converting to a chemical, it simply skips the converting process and does it.

# TODO

- [ ] The Balancing Part
- [x] The Coding Part

# Changelog

:cl:
- add: Added Succubus Trait

Raikyr0 committed:
Add files via upload

Raikyr0 committed:
Update MetabolizerTypes.yml

Raikyr0 committed:
Update MetabolizerTypes.yml

Raikyr0 committed:
Update metabolizer-types.ftl

Raikyr0 committed:
Update lewd.yml

Raikyr0 committed:
Update traits.ftl

Raikyr0 committed:
Update natural_sauce.yml

Raikyr0 committed:
Update natural_sauce.yml

Raikyr0 committed:
Update natural_sauce.yml

Raikyr0 committed:
Update SuccubusSystem.cs

Raikyr0 commented:
Just running the checks before putting it back into draft for balancing and bug fixing. (Surely the build and test debug will show me the actual issues instead of failing 62 times to unrelated issues)

Raikyr0 committed:
Update lewd.yml

Raikyr0 committed:
Delete Content.Server/_Floof/Traits/SuccubusSystem.cs

Raikyr0 committed:
Delete Content.Server/_Floof/Traits/SuccubusComponent.cs

Raikyr0 committed:
Add files via upload

Raikyr0 committed:
Merge branch 'Fansana:master' into CumSuckingFiend

Raikyr0 committed:
Update lewd.yml

M3739 commented:
Alright, so I tested it out a bit, and it appears that it only seems to result in a blood level increase, but no healing anywhere to be found.

The saline part is coming through, and so is the vitamins, but the omnizine portion is not coming up.

Raikyr0 commented:
> Alright, so I tested it out a bit, and it appears that it only seems to result in a blood level increase, but no healing anywhere to be found.
> 
> The saline part is coming through, and so is the vitamins, but the omnizine portion is not coming up.

that's really weird, maybe there's too many chemicals being processed?

Raikyr0 committed:
Update medicine.ftl

Raikyr0 committed:
Update natural_sauce.yml

Raikyr0 committed:
Update drugs.yml

Raikyr0 committed:
Update drugs.yml

Raikyr0 committed:
Update mood.ftl

Raikyr0 committed:
Update medicine.yml

Raikyr0 committed:
Update medicine.yml

Raikyr0 committed:
Update mood.ftl

Raikyr0 committed:
Update mood.ftl

Raikyr0 committed:
Update drugs.yml

Raikyr0 committed:
Update drugs.yml

Raikyr0 committed:
Update MetabolizerTypes.yml

Raikyr0 commented:
Ok, I added a chemical called Succubium, (tad on the nose I know) which combines the effects of the 3 chemicals so it doesn't clog the bloodstream.

M3739 commented:
Gonna test this again later on.

M3739 reviewed (changes requested):
After some testing,  it now doesn't seem to work at all. No longer raises blood levels.
![image](https://github.com/user-attachments/assets/558288f0-40ba-49d5-aa96-01f5267ad6b8)

Raikyr0 commented:
> After some testing, it now doesn't seem to work at all. No longer raises blood levels. ![image](https://private-user-images.githubusercontent.com/47579354/457535168-558288f0-40ba-49d5-aa96-01f5267ad6b8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTA0NjQ5NjAsIm5iZiI6MTc1MDQ2NDY2MCwicGF0aCI6Ii80NzU3OTM1NC80NTc1MzUxNjgtNTU4Mjg4ZjAtNDBiYS00OWQ1LWFhOTYtMDFmNTI2N2FkNmI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjIxVDAwMTEwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcxMDk2YWE3NzEyOTQ4MDg0YjY5NmU1ZjU1MzQ0OTE0MThkYTA3NjMzNjhiMTcwZTM0YjFmMGMwNjJiZmNjMWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.aKAaj484rlzoMfDa9MFOH6MZmEbIzeWFKEL1lN7xknY)

Not sure what's wrong, could you inject the succubus chemical instead of cum? It's either the chemical that's not working. Or the conversion process.

Raikyr0 committed:
Update medicine.yml

realized I pumped the food/drink numbers for succubus up on cum so it doesn't really need it here

M3739 commented:
> Not sure what's wrong, could you inject the succubus chemical instead of cum? It's either the chemical that's not working. Or the conversion process.

Ill test that next.

M3739 commented:
> Ill test that next.

The Succubium works as intended. Pretty good at raising blood levels, actually. It only heals ever so slightly.
![image](https://github.com/user-attachments/assets/a948a14e-e63b-41d9-ab69-f5ed6ed9b23e)

Raikyr0 commented:
> > Ill test that next.
> 
> The Succubium works as intended. Pretty good at raising blood levels, actually. It only heals ever so slightly. 

Yeah it's not supposed to heal a lot.

Raikyr0 committed:
Update natural_sauce.yml

Raikyr0 committed:
Update natural_sauce.yml

Raikyr0 committed:
Update medicine.yml

Raikyr0 committed:
Update medicine.ftl

Raikyr0 committed:
Merge branch 'master' into CumSuckingFiend

M3739 reviewed (changes requested):
Still does not work, unfortunately...
![image](https://github.com/user-attachments/assets/33afbc53-b793-4d78-8cc8-49ac1d3f465d)
Here's a bit of a pointer. the `!type:OrganType` condition looks for a `MetabolizerType`  in a metabolizing organ.

That being said, we have an in-house trait modify type called `!type:TraitModifyMetabolism`, courtesy of @Superlagg and his #707 that may be useful to achieve your ends. You can read more about it in this file located [here](https://github.com/Fansana/floofstation1/blob/0e86f73bda61ae9a2a3ab7177b4b882babb7550c/Content.Server/_Floof/Traits/TraitSystem.Functions.cs#L16-L37).

Raikyr0 committed:
Update lewd.yml

Raikyr0 commented:
> Still does not work, unfortunately... ![image](https://private-user-images.githubusercontent.com/47579354/458770968-33afbc53-b793-4d78-8cc8-49ac1d3f465d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTA4NDM0OTksIm5iZiI6MTc1MDg0MzE5OSwicGF0aCI6Ii80NzU3OTM1NC80NTg3NzA5NjgtMzNhZmJjNTMtYjc5My00ZDc4LThjYzgtNDlhYzFkM2Y0NjVkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjI1VDA5MTk1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA3NzNhZGY4ZGZiOWVlYTE5ZWU1MDA2MWUzYTlkYTFlMmY4YWQ1Zjk3Y2VhYzVjZmNiMzU4NTIzMTg5MWJjNTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.HOPrgoVjCBk9NxqvoFX2iKjDPjVWRQ2-IWQprBYHz7Y) Here's a bit of a pointer. the `!type:OrganType` condition looks for a `MetabolizerType` in a metabolizing organ.
> 
> That being said, we have an in-house trait modify type called `!type:TraitModifyMetabolism`, courtesy of @Superlagg and his #707 that may be useful to achieve your ends. You can read more about it in this file located [here](https://github.com/Fansana/floofstation1/blob/0e86f73bda61ae9a2a3ab7177b4b882babb7550c/Content.Server/_Floof/Traits/TraitSystem.Functions.cs#L16-L37).

I'll try the  - !type:TraitModifyMetabolism if it works I might have to clean up the excess files

M3739 reviewed (changes requested):
Still doesn't work. Perhaps try adding the `add: true` field to the `TraitModifyMetabolism`?

Raikyr0 committed:
Update lewd.yml

M3739 reviewed (commented):
Happy to report it works as intended following the latest commit.

Just do some clean up, then we can discuss balancing.

Raikyr0 committed:
Delete Content.Server/_Floof/Traits/SuccubusComponent.cs

Raikyr0 committed:
Delete Content.Server/_Floof/Traits/SuccubusSystem.cs

M3739 reviewed (approved):
Functionality is confirmed, and it's working as intended.

With how scarce the actual thing is in comparison to blood, I'd reckon it's okay in it's current state. While allowing for some interesting scenarios.

Ill invite discourse from the other maintainers to ensure the balancing part, as many pairs  of eyes are better than one.

Mnemotechnician reviewed (approved):
i don't mind this so long as it actually works. However, one thing to look out for is the possibility of people powergaming with this trait by e.g. constantly carrying a bottle-of-this-thing and chugging from it to regenerate, but I guess that also applies to vampires.

Raikyr0 committed:
Merge branch 'master' into CumSuckingFiend

Commit:
Merge branch 'master' into CumSuckingFiend

Raikyr0 committed:
Merge branch 'master' into CumSuckingFiend

M3739 commented:
Bruhg

M3739 commented:
Alright, this has been sitting for a long while. Merging.

ashkitten commented:
> one thing to look out for is the possibility of people powergaming with this trait by e.g. constantly carrying a bottle-of-this-thing and chugging from it to regenerate, but I guess that also applies to vampires.

Something that could be considered for the future is adding DNA to more bodily fluids than just blood, and then having the trait only work if there's DNA present. However, I suspect that would not be appreciated by some people who want more "vegan" characters.
